code/modules/mob/living/carbon/human/ai/action_datums/throw_grenade.dm

```diff
@@ -5,7 +5,10 @@
 	var/mid_throw
 
 /datum/ai_action/throw_grenade/get_weight(datum/human_ai_brain/brain)
-	if(!brain.in_combat || !brain.grenading_allowed)
+	if(!brain.grenading_allowed)
+		return 0
```

Comment:
why split this out?

Reply:
For general readability - for me, it's easier to keep track of conditions when they are split between lines, rather than one long if statement. Basically speaking - we have no real reason to do it